src/ECL/ECL.hs

```diff
@@ -31,6 +35,15 @@ data Binder lit
   | Record [(Name, Binder lit)]
   deriving (Show, Eq)
 
+-- | Represents constructors' arities.
+type Arity = Int
+
+-- | Environment
+--
+-- The language implementor should provide an environment to let
+-- the checker lookup for constructors' names (just for one type for now).
+type Environment = [(Name, Arity)]
```

Comment:
@paf31 I also think we should have something like:

```haskell
newtype Environment = Environment { envArities :: [Name -> [(Name, Arity)]] }
```
or
```haskell
newtype Environment = Environment { envArities :: Name -> (Type, [(Name, Arity)]) }
```
I'm thinking of a representation that will let us, knowing just the name of one constructor, obtain the rest of constructors of that type with their arities.